### ISSUE TITLE
admin_columns: some timestamps not being converted correctly

### ISSUE THREAD

**Issue by lumpysimon** [johnbillion/extended-cpts#105]
I'm saving dates in a meta field in Unix timestamp format and showing them in an admin column with the following code:

```
'admin_cols' => array(
	'document-date' => array(
		'title'       => 'Date',
		'meta_key'    => 'document-date',
		'date_format' => get_option( 'date_format' ),
		'default'     => 'DESC'
	)
)
```

The date format option is set to `F j, Y`

Here's the output, the 3rd and 4th dates are incorrect: 
<img width="775" alt="screen shot 2018-04-05 at 18 04 12" src="https://user-images.githubusercontent.com/625372/38380492-cf8a271a-38fb-11e8-8015-2c36133e81f1.png">

The timestamps for the two messed up ones are:

1522368000 (12am 30th March 2018)
1520380800 (12am 7th March 2018)

I can't figure out why some dates are ok and some are not.

I'm running PHP 7.1.4

**Comment by lumpysimon** [comment]
I'm seeing this same issue on another site, so I took a look at the code, it looks like lines 757-761 in `class-extended-cpt-admin.php` are the culprits:

```
$val_time = strtotime( $val );
if ( $val_time ) {
	$val = $val_time;
}
```

If I remove those lines the dates are displayed correctly. However, I'm not sure of any unintended side effects of doing this? I've tested with dates stored as Unix timestamp and `YYYY-MM-DD HH:MM:SS` and both work without those lines.

**Comment by johnbillion** [comment]
This was caused by #77, but I'm not sure why some timestamps would not be interpreted correctly by `strtotime()`.

**Comment by davekellam** [comment]
I ran into this today on a client site, can confirm that it's an issue and the problem is related to #77. 

It appears that some values of epoch time return "valid" `strtotime()` values. For me, `1607518800` evaluated to `215559850071` and `1607524200` evaluated to `70397827672` resulting in some _way_ off times. Happened for other results as well, but this was the first I noticed.

<img width="534" alt="Screen Shot 2020-10-26 at 3 54 32 PM" src="https://user-images.githubusercontent.com/60804/97249853-1571fa80-17db-11eb-97e2-3dba369e8110.png">

Was able to replicate this with `4.4.0`, it seems that for epoch time to be converted to epoch time properly, you need to add an `@`at the start. See https://www.php.net/manual/en/datetime.formats.compound.php

So, my initial attempt might be overly complex, but seemed to give me the right results. Albeit, I've done _no_ testing or anything outside the unix timestamp (like to see if other strings work fine). 

```php
try {
	new DateTime ( '@' . $val );
	$val_time = strtotime( '@' . $val );
} catch ( Exception $e ) {
	$val_time = strtotime( $val );
}
```

The idea being, check if it's a valid timestamp, if not try `strtotime`, and then carry on. If this seem alright, I can pull down the repo, modify the code, and submit a pull request. Open to other approaches!

**Comment by cgarofalo** [comment]
I have tried @davekellam 's solution and can confirm that it works.
Insert the snippet on line 738 in `class-extended-cpt-admin.php`

This would be the whole function:

```php
   /**
	 * Output column data for a post meta field.
	 *
	 * @param string $meta_key The post meta key
	 * @param array  $args     Array of arguments for this field
	 */
	public function col_post_meta( string $meta_key, array $args ) {

		$vals = get_post_meta( get_the_ID(), $meta_key, false );
		$echo = [];
		sort( $vals );

		if ( isset( $args['date_format'] ) ) {

			if ( true === $args['date_format'] ) {
				$args['date_format'] = get_option( 'date_format' );
			}

			foreach ( $vals as $val ) {
				$val_time = strtotime( $val );

				try {
					new DateTime ( '@' . $val );
					$val_time = strtotime( '@' . $val );
				} catch ( Exception $e ) {
					$val_time = strtotime( $val );
				}
				
				if ( $val_time ) {
					$val = $val_time;
				}

				if ( is_numeric( $val ) ) {
					$echo[] = date_i18n( $args['date_format'], $val );
				} elseif ( ! empty( $val ) ) {
					$echo[] = mysql2date( $args['date_format'], $val );
				}
			}
		} else {

			foreach ( $vals as $val ) {

				if ( ! empty( $val ) || ( '0' === $val ) ) {
					$echo[] = $val;
				}
			}
		}

		if ( empty( $echo ) ) {
			echo '&#8212;';
		} else {
			echo esc_html( implode( ', ', $echo ) );
		}

	}
```

**Comment by johnbillion** [comment]
Thanks for working on this everyone. Fixed in #187.
